qiskit/algorithms/optimizers/gradient_descent.py

```diff
@@ -191,11 +196,11 @@ def __init__(
                 approximated with a forward finite difference scheme with ``perturbation``
                 perturbation in both directions (defaults to 1e-2 if required).
                 Ignored when we have an explicit function for the gradient.
+            new_callback_signature: use new callback signature (``OptimizerCallback``).
```

Comment:
What do you think about setting this in the `algorithm_globals` instead of in each algorithm? Or we could make a future import like
```
from qiskit.future import optimizer_callbacks
```
to enable the new functionality.